ibis/bigquery/client.py

```diff
@@ -219,4 +218,9 @@ def _discover_type(field):
 
 def bigquery_table_to_ibis_schema(table):
     pairs = ((el.name, _discover_type(el)) for el in table.schema)
+    try:
+        if table.list_partitions():
```

Comment:
Can you use my original query here so that we don't pull down a ton of partitions?

Reply:
there's at most 2500 so I don't think its burdensom, this way there's no need poke around with their internal representation